src/WorkerServiceProvider.php

```
@@ -75,21 +76,24 @@ public function boot()
                         $status = "exception";
                         $errorCode = isset($event->exception) ? $event->exception->getCode() : -1;
                         break;
+                    default:
+                        $status = "unkown_status";
+                        $errorCode = -1;
                 }
 
                 $histogram = app('prometheus.workers.client.histogram');
                 $histogram->observe(
                     microtime(true) - self::$starts,
                     [
-                        $event->job->resolveName(),
-                        $event->job->getQueue(),
-                        $event->connectionName,
+                        self::$jobName,
+                        self::$queueName,
+                        self::$connectionName,
                         $errorCode,
                         $status,
                     ]
                 );
-            } catch (\Throwable $e) {
-                //fail silently
+            } catch(\Throwable $e) {
+                // fail silently and don't stop the application
```

@m-abdelmohsen Any idea what are the chances reaching here?

---

I think we can fail silently but, clearly, we need to report this somewhere else how we'll see it if something goes wrong here? At least we should have a log here.

Following what Shameer said, are we 100% we should fail silently here? Maybe is worthy to fail and let the worker restart, in any case this must be reported IMO.

---

+1 for the log message here

---

Added a log message with the error + reported to bugsnag